### DISCUSSION TITLE
[Gatsby Cloud] Crashes due to babel warning

### DISCUSSION THREAD

**Discussion by aidanm1999** [gatsbyjs/gatsby#36870] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

With medium to large websites built on Gatsby an error appears: 

```zsh
 ERROR 

[BABEL] Note: The code generator has deoptimised the styling of /.cache/page-ssr/routes/render-page.js as it exceeds the max of 500KB.
```

This usually is a 

### Reproduction Link

https://codesandbox.io/s/gatsby-cloud-babel-crash-o0yo9b

### Steps to Reproduce

1. Go to the [Reproduction Link](https://codesandbox.io/s/gatsby-cloud-babel-crash-o0yo9b) which has been forked from the [`gatsby-starter-minimal`](https://githubbox.com/gatsbyjs/gatsby-starter-minimal) CodeSandbox. Changes:
    - Added a component with 10K lines of text
    - Added `mui` libraries
    - Added the `gatsby-node.js` page to generate SSR pages
2. Open a new terminal within the sandbox and run `yarn build`
3. Look at the output

### Expected Result

During the build process, you should see:

```zsh
[...]
success Building Rendering Engines - 50.016s
success Building HTML renderer - 28.971s
success Execute page configs - 7.143s
success Validating Rendering Engines - 3.884s
success Caching Webpack compilations - 0.002s
success run queries in workers - 0.095s - 1/1 10.48/s
success Merge worker state - 0.003s
[...]
```

### Actual Result

```zsh
[...]
success Building Rendering Engines - 50.016s
success Building HTML renderer - 28.971s

 ERROR 

[BABEL] Note: The code generator has deoptimised the styling of /.cache/page-ssr/routes/render-page.js as it exceeds the max of 500KB.

success Execute page configs - 7.143s
success Validating Rendering Engines - 3.884s
success Caching Webpack compilations - 0.002s
success run queries in workers - 0.095s - 1/1 10.48/s
success Merge worker state - 0.003s
[...]
```

### Environment

```shell
System:
    OS: Linux 5.4 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (16) x64 Intel(R) Core(TM) i9-9900K CPU @ 3.60GHz
    Shell: 5.0.3 - /bin/bash
  Binaries:
    Node: 14.19.3 - /tmp/yarn--1666095923066-0.8819107191859694/node
    Yarn: 1.22.19 - /tmp/yarn--1666095923066-0.8819107191859694/yarn
    npm: 6.14.17 - ~/.nvm/versions/node/v14.19.3/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    gatsby: ^4.24.4 => 4.24.4
```

### Config Flags

None set

# Current debug process

When looking at the Babel Docs, the option that looks like it would turn this off is [`compact`](https://babeljs.io/docs/en/options#compact) which wouldn't throw this issue:

> Type: boolean | "auto"
> Default: "auto"
> 
> "auto" will set the value by evaluating code.length > 500_000
> 
> All optional newlines and whitespace will be omitted when generating code in compact mode.

On the [Gatsby site's Babel page](https://www.gatsbyjs.com/docs/how-to/custom-configuration/babel/), it mentions creating a `.babelrc` and adding [`babel-preset-gatsby`](https://github.com/gatsbyjs/gatsby/tree/master/packages/babel-preset-gatsby). Looking at `babel-preset-gatsby`, the only available `options` are `targets`, `reactRuntime` and `reactImportSource` so I don't think that would work. Below are some `compact` options I have tried within the `.babelrc`:

```json
{
    "presets": [
        [
            "@babel/preset-react",
            {
                "options": {
                    "compact": false
                },
                "compact": false,
                "overrides": [
                    {
                        "test": "./.cache/page-ssr/routes/render-page.js",
                        "compact": false
                    },
                    {
                        "test": "*",
                        "compact": false
                    }
                ]
            }
        ]
    ],
    "compact": false,
    "overrides": [
        {
            "test": "./.cache/page-ssr/routes/render-page.js",
            "compact": false
        },
        {
            "test": "*",
            "compact": false
        }
    ]
}
```

I also saw that Gatsby Node exposes `onCreateWebpackConfig` and `onCreateBabelConfig` which exposes `setBabelOptions` so I have tried calling that function:

```ts
export const onCreateBabelConfig = ({ stage, actions }) => {
    actions.setBabelOptions({
        compact: false,
        options: {
            compact: false,
            generatorOpts: {
                compact: false,
            },
        },
    });
};
```

```ts
export const onCreateWebpackConfig = ({ stage, actions }) => {
    actions.setBabelOptions({
        compact: false,
        options: {
            compact: false,
            generatorOpts: {
                compact: false,
            },
        },
    });
};
```

**Comment by LekoArts** [comment]
It's an error that is thrown and shown in the CLI but it's not one that actually crashes the build. I'm not sure right why this wouldn't work:

```
{
  "presets": ["babel-preset-gatsby"]
  "compact": false,
}
```

At least as a starting point to see if it does anything. Afterwards one could look at `overrides`

**Reply by aidanm1999** [reply]
Hi @LekoArts,

Thanks for the reply, unfortunately replacing the contents of `.babelrc` and installing `babel-preset-gatsby` with the above still outputs the babel error.

**Reply by aidanm1999** [reply]
Hi @LekoArts,

Is this not also a bug with Gatsby Cloud as the build fails? I tried bringing this up with Gatsby Cloud but it said I needed to have a paid account. The build works fine on other platforms like Vercel and Netlify, just not Gatsby Cloud.

**Comment by aidanm1999** [comment, marked as answer]
For anyone having this issue, a clean restart of the cloud instance solved the issue. The babel issue was a red herring.
